### PR TITLE
Avoid PTR verification when provisioning in Enclave

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java [hunk 1]
```diff
@@ -73,9 +76,23 @@ public Set<String> resolveAll(String name) {
 
     @Override
     public Set<String> resolve(String name, RecordType first, RecordType... rest) {
-        return resolveAll(name);
+        Set<RecordType> types = new HashSet<>(1 + rest.length);
+        types.add(first);
+        Collections.addAll(types, rest);
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java:81]: Use `EnumSet.of(first, rest)` ?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java [hunk 2]
```diff
@@ -73,9 +76,23 @@ public Set<String> resolveAll(String name) {
...
+                .stream()
+                .filter(addressString -> {
+                    if (types.contains(RecordType.A) && IP.isV4(addressString)) return true;
+                    if (types.contains(RecordType.AAAA) && IP.isV6(addressString)) return true;
+                    return false;
+                })
+                .collect(Collectors.toSet());
     }
 
+    private static boolean isIpv6(String ipAddress) { return ipAddress.contains(":"); }
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java:93]: Not used?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java [hunk 3]
```diff
@@ -415,6 +426,27 @@ private static Allocation ofIpv4(String ipAddress, NameResolver resolver) {
             return new Allocation(hostname4, Optional.of(addresses.get(0)), Optional.empty());
         }
 
+        private static Allocation fromAddress(Address address, NameResolver resolver, IpAddresses.Protocol protocol) {
+            Optional<String> ipv4Address = resolveOptional(address.hostname(), resolver, RecordType.A);
+            if (protocol != IpAddresses.Protocol.ipv6 && ipv4Address.isEmpty())
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java:431]: Avoid resolving `A` records in the first place if protocol is `IPv6`?

**Review by freva — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java:81]: Great, exactly what I was looking for.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/testutils/MockNameResolver.java:93]: Removed

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java (hunk 3 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java:431]: :thinking: or, verify no A records are found for a supposedly IPv6-only host?  We expect later resolutions in host admin or various tools to blindly find all A and AAAA records, so let's ensure they are exactly as expected?  Implemented the latter.

**Review by freva — APPROVED** [review]

**Pull request merged**